Add spaces to speak trigger

lukec11 opened the pull request:
It currently activates upon words like "hearsay," "essay", "retreat" which don't relate to the bot but match it by technicality

lukec11 committed:
Add spaces to speak trigger

It currently activates upon words like "hearsay," "essay", "retreat" which don't relate to the bot but match it by technicality

maxwofford commented:
I'd recommend switching to regex for this. I don't think the current implementation would respond to "maildog speak"

lukec11 commented:
Good idea, didn't want to remove the ease of using a default js array but it's not significantly harder in this case to deal with a regex non-capture group I suppose

lukec11 committed:
Use regex for detection

It only works if there is a non-word character beforehand (such as a space or number) or the message begins with the trigger
